packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -125,32 +59,35 @@ function isAction<A extends AnyAction>(
   return action.type === type;
 }
 
-export type QueryBarFormField = {
-  value: unknown;
-  string: string;
-  valid: boolean;
-};
-
 export type QueryBarState = {
-  fields: Record<QueryProperty, QueryBarFormField>;
+  fields: QueryFormFields;
   expanded: boolean;
   serverVersion: string;
   schemaFields: unknown[];
-  lastAppliedQuery: unknown | null;
+  lastAppliedQuery: BaseQuery | null;
   /**
    * For testing purposes only, allows to track whether or not apply button was
    * clicked or not
    */
   applyId: number;
+  activeTab: QueryHistoryTab;
+  namespace: string;
+  host?: string;
+  recentQueries: RecentQuery[];
+  favoriteQueries: FavoriteQuery[];
 };
 
 export const INITIAL_STATE: QueryBarState = {
-  fields: mapQueryToValidQueryFields(DEFAULT_FIELD_VALUES),
+  fields: mapQueryToFormFields(DEFAULT_FIELD_VALUES),
   expanded: false,
   serverVersion: '3.6.0',
   schemaFields: [],
   lastAppliedQuery: null,
   applyId: 0,
+  activeTab: 'recent',
```

Comment:
Any reason for this to be in store state and not in the QueryHistory component state?

Reply:
Ahhh, got it, so we can switch to the favorites when query is saved

Reply:
I wanted it there, but when we save recent query as favorite, we navigate user to favorites tab and for that reason added it here.

Reply:
actually I can return a value from the action (saveRecentAsFavorite) and then change it in the component.

Reply:
I think it's okay if you want to keep it in the state as the state is somewhat relevant to the non-ui logic, but this pattern with returning from action is also something that will fit here, yeah

Reply:
added in [e5443c0](https://github.com/mongodb-js/compass/pull/4585/commits/e5443c03ed3ae3ad25e165d1a317656caed0a4b4)